mathesar_ui/src/component-library/common/styles/components/input.scss

```diff
@@ -23,10 +23,6 @@
     background: var(--slate-100);
   }
 
-  &:read-only {
-    background: var(--slate-50);
-  }
-
```

Comment:
Removing the CSS will negate any style modifications if the component has the 'readonly' prop enabled